Sming/SmingCore/ArduinoCompat.h

```diff
@@ -25,5 +25,15 @@ void yield();
 }
 #endif
 
+#define abs(x)                         ((x)>0?(x):-(x))
```

Comment:
I tried them there, but I had trouble getting things to compile.  It would be good to get rid of those macros altogether but I understand we need to support the older style Arduino code.  Happy to see them move to a better place.